activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -459,6 +443,37 @@ def serialize_entries(entries, **options)
           end
         end
 
+        def change_counter(key, amount, options)
+          redis.then do |c|
+            c = c.node_for(key) if c.is_a?(Redis::Distributed)
+
+            if options[:expires_in] && supports_expire_nx?
+              c.pipelined do |pipeline|
+                pipeline.incrby(key, amount)
+                pipeline.call(:expire, key, options[:expires_in].to_i, "NX")
+              end.first
+            else
+              count = c.incrby(key, amount)
+              if count != amount && options[:expires_in] && c.ttl(key) < 0
+                c.expire(key, options[:expires_in].to_i)
+              end
+              count
+            end
+          end
+        end
+
+        def supports_expire_nx?
+          return @supports_expire_nx if defined?(@supports_expire_nx)
+          @supports_expire_nx = redis_version >= Gem::Version.new("7.0.0")
+        end
+
+        def redis_version
+          @redis_version ||= begin
+            version = redis.then { |c| c.info("server").fetch("redis_version") }
+            Gem::Version.new(version)
+          end
+        end
```

Comment:
This is only used in `supports_expire_nx?`, so I don't think it's work memoizing the Version object. I would inline it in `supports_expire_nx?`

Reply:
Inlined.